src/serialization/string_utils.cpp

```diff
@@ -822,57 +823,102 @@ std::vector<std::string> quoted_split(const std::string& val, char c, int flags,
 	return res;
 }
 
+namespace
+{
+/**
+ * Internal common code for parse_range and parse_range_real.
+ *
+ * If str contains two elements and a separator such as "a-b", returns a and b.
+ * Otherwise, returns the original string and std::nullopt.
+ */
+std::pair<std::string, std::optional<std::string>> parse_range_internal_separator(const std::string& str)
```

Comment:
While it could do, any string returned is expected to fit within the short-string optimisation length. I think it'll turn out to be about the same amount of data either way round, even before the compiler optimisations.

Reply:
> While it could do, any string returned is expected to fit within the short-string optimisation length. I think it'll turn out to be about the same amount of data either way round, even before the compiler optimisations.


Hmm, i really don't know how much the impact is then, i think this is probably something only an actual benchmark will tell us, (which we probably won't do).


>because it can be implemented based on a C function that expects the null-terminator at the end.

From how i understood it, it calls std::strtol which doesn't expect a terminating null character, 

https://en.cppreference.com/w/cpp/string/basic_string/stol

Reply:
(There's a bit of discussion on IRC after this)

@gfgtdf would you accept the PR as-is? I think any changes to use `from_chars` should be a separate PR.

Reply:
I didnt check the whole pr but the comment above is not something that i'd require to be solved before merging, especially since it seems like that creating a std::string was also the beahvior before